src/App.jsx

```diff
@@ -14,57 +16,67 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+export const App = () => {
+  const [goodSelected, setGoodSelected] = useState(goods[goods.length - 2]);
+  const goodSelectedTitle = `${goodSelected} is selected`;
+  const titleClearButton = (
+    <button
+      data-cy="ClearButton"
+      type="button"
+      className="delete ml-3"
+      onClick={() => setGoodSelected('')}
+    />
+  );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  function isSelected(good) {
+    return good === goodSelected;
+  }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  function handleGoodButtonClick(good) {
+    if (isSelected(good)) {
+      setGoodSelected('');
+    } else {
+      setGoodSelected(good);
+    }
+  }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {goodSelected ? goodSelectedTitle : 'No goods selected'}
+        {goodSelected && titleClearButton}
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map(good => {
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={
+                  isSelected(good) ? 'has-background-success-light' : ''
```

Comment:
You don't need `isSelected()` function because you don't transmit any other parametres but `good`, so you can you put it on a separate variable at the beginning of your map method before `return`. Calling a function is just an extra operation

Reply:
Interesting suggesting...